### PR TITLE
Rook persistent storage for a kubernetes cluster

### PR DESCRIPTION
**Pull request by travisn**

### CODE REVIEW

File: multi-node/generic/controller-install.sh [hunk 1]
```diff
@@ -84,6 +87,49 @@ function init_flannel {
 }
 
 function init_templates {
+    local CEPH_ADMIN_SECRET=AQCE9hxYbWb5GBAA/QaOJw7ruR/NUFPKwy66QA==
```
Comment (bassamtabbara) [inline multi-node/generic/controller-install.sh]: a comment here would be good

File: multi-node/generic/controller-install.sh [hunk 2]
```diff
@@ -84,6 +87,49 @@ function init_flannel {
...
+  --mount volume=var-lib-rook,target=/var/lib/rook \
+  $RKT_OPTIONS \
+  $ROOKD_ACI
+
+[Install]
+WantedBy=multi-user.target
+RequiredBy=kubelet.service
+EOF
+    fi
+
```
Comment (bassamtabbara) [inline multi-node/generic/controller-install.sh]: this should all go into a rookd-wrapper at some point, right?

File: multi-node/generic/controller-install.sh [hunk 3]
```diff
@@ -84,6 +87,49 @@ function init_flannel {
...
+ExecStartPre=/usr/bin/mkdir -p /var/lib/rook
+ExecStart=/usr/bin/rkt run \
+  --trust-keys-from-https \
+  --stage1-from-dir=stage1-fly.aci \
+  --inherit-env \
+  --volume dns,kind=host,source=/run/systemd/resolve/resolv.conf,readOnly=true \
+  --mount volume=dns,target=/etc/resolv.conf \
+  --volume var-lib-rook,kind=host,source=/var/lib/rook \
+  --mount volume=var-lib-rook,target=/var/lib/rook \
```
Comment (bassamtabbara) [inline multi-node/generic/controller-install.sh]: where is /var/lib/rook created?

File: single-node/rbd.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+                    }
+                ]
+            }
+        ],
+        "volumes": [
+            {
+                "name": "rbdpd",
+                "rbd": {
+                    "monitors": [
+        						"172.17.4.99:6790"
```
Comment (thephred) [inline single-node/rbd.json]: Too many tabs?

**Pull request merged**